pkg/application/application.go

```diff
@@ -288,48 +288,50 @@ func newKustApp(o *CreateOptions, projectName, repoURL, targetRevision, repoRoot
 	return app, nil
 }
 
-func (app *kustApp) CreateFiles(repofs fs.FS, projectName string) error {
-	return kustCreateFiles(app, repofs, projectName)
+func (app *kustApp) CreateFiles(repofs fs.FS, appsfs fs.FS, projectName string) error {
+	return kustCreateFiles(app, repofs, appsfs, projectName)
 }
 
-func kustCreateFiles(app *kustApp, repofs fs.FS, projectName string) error {
+func kustCreateFiles(app *kustApp, repofs fs.FS, appsfs fs.FS, projectName string) error {
 	var err error
 
 	// create Base
-	appPath := repofs.Join(store.Default.AppsDir, app.Name())
-	basePath := repofs.Join(appPath, "base")
-	baseKustomizationPath := repofs.Join(basePath, "kustomization.yaml")
+	appPath := appsfs.Join(store.Default.AppsDir, app.Name())
+	basePath := appsfs.Join(appPath, "base")
+	baseKustomizationPath := appsfs.Join(basePath, "kustomization.yaml")
 
 	// check if base is in the same filesystem
-	if repofs.ExistsOrDie(baseKustomizationPath) {
+	if appsfs.ExistsOrDie(baseKustomizationPath) {
 		// check if the bases are the same
 		log.G().Debug("application base with the same name exists, checking for collisions")
-		if collision, err := checkBaseCollision(repofs, baseKustomizationPath, app.base); err != nil {
+		if collision, err := checkBaseCollision(appsfs, baseKustomizationPath, app.base); err != nil {
 			return err
 		} else if collision {
 			return ErrAppCollisionWithExistingBase
 		}
+	} else if appsfs != repofs && repofs.ExistsOrDie(appPath) {
+		return ErrAppBaseOnDifferentRepo
```

Comment:
don't know why I can't comment on the error definition itself but I think it should be more precise, not 
```
an application with the same name already exists in a different repo, consider choosing a different name
```
but 
```
an application with the same base exists on the bootstrap repo (or whatever we call the main repo)
```